keps/sig-storage/1495-volume-populators/README.md

```diff
@@ -271,6 +294,153 @@ downgrade.
 
 No issues
 
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [X] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: AnyVolumeDataSource
+    - Components depending on the feature gate: kube-apiserver
+  - [ ] Other
+    - Describe the mechanism:
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+
+* **Does enabling the feature change any default behavior?**
+  Before this feature, kube API server would silently drop any PVC data source
+  that it didn't recognize, as if the client never populated the field. After
+  enabling this feature, the API server allows any object to be specified.
+  An external admission controller will take over validation of the field,
+  and its response to invalid data sources will be to fail the create operation,
+  rather than silently drop them.
```

Comment:
The external admission controller is an optional component, right?  Can you describe what will happen if it is not installed?  Can you also describe who should be deploying the new components, i.e., the external admission controller and the new CRDs?

Reply:
I think it would be bad to set the feature gate on and to not install this controller. For that reason I'm suggesting we leave this in alpha until the webhook actually exists and people can test it, and we only move to beta once that approach is accepted.

I will add the detail you requested.